internal/pkg/logging/workload.go

```diff
@@ -173,7 +173,9 @@ func (s *WorkloadClient) WriteLogEvents(opts WriteLogEventsOpts) error {
 		LogStreamLimit:      opts.LogStreamLimit,
 	}
 
-	logEventsOpts.LogStreamPrefixFilters = s.logStreams(opts.TaskIDs)
+	if s.logStreamNamePrefix != "" {
```

Comment:
Since app runner client is the only workload client that don't have `logStreamNamePrefix` set up: if `logStreamNamePrefix == "" ➡️  it's app runner ➡️  don't filter by `copilot` prefix.

I think this totally makes sense as a quick workaround! However we should probably refactor it later so that it works without assuming `logStreamNamePrefix == "" ➡️  it's app runner`. This is effectively true but is not necessarily true in terms of logic, which may make it confusing to read and maintain in the future. Do you think we want to add a `TODO` comment on this, explain why there is a `if` block and what we need to do about it in the future?

Reply:
great point - i added a TODO saying what this does along with @efekarakus's suggestion to have a separate logging client for App Runner services.